### PR TITLE
feat(widget-chat): deleted message support in activity list

### PR DESCRIPTION
**Pull request by adamweeks**
This will display a system message for activities that have been deleted upon initial load. It also will convert existing messages to a deleted message when an activity arrives via mercury.

### CODE REVIEW

File: packages/widget-chat/src/actions/conversation.js [hunk 1]
```diff
@@ -58,6 +66,9 @@ export function listenToMercuryActivity(conversationId, spark) {
     spark.mercury.on(`event:conversation.activity`, (event) => {
       const activity = event.data.activity;
       if (activity.object.objectType === `comment` && activity.target.id === conversationId) {
```
Comment (adamweeks) [inline packages/widget-chat/src/actions/conversation.js]: Good point, I blame @bzang 😛

File: packages/widget-chat/src/actions/conversation.js [hunk 2]
```diff
@@ -58,6 +66,9 @@ export function listenToMercuryActivity(conversationId, spark) {
...
+        dispatch(receiveMercuryComment(activity));
+      }
+      else if (activity.object.objectType === `activity` && activity.target.id === conversationId) {
```
Comment (adamweeks) [inline packages/widget-chat/src/actions/conversation.js]: Interesting, we hadn't encountered a lot of different types yet, so this will be good to add.

Comment (ianwremmel) [inline packages/widget-chat/src/actions/conversation.js]: yea, it's not obvious it's necessary, but if we don't add the explicit checks now, we're going to end up with weird breakages in the future. the idea is basically: check all conditions and if there's no match, discard the activity. better to call attention to not-yet-implemented features that break catastrophically (which has happened when verbs get overloaded).

File: packages/widget-chat/src/components/activity-item/index.test.js [hunk 3]
```diff
@@ -26,3 +26,27 @@ describe(`ActivityItem component`, () => {
...
+    verb: `tombstone`
+  };
+
+  const component = renderer.create(
+    <ActivityItem
+      content={activity.content}
+      id={activity.id}
+      name={activity.name}
+      timestamp={activity.timestamp}
+      verb={activity.verb}
```
Comment (bzang) [inline packages/widget-chat/src/components/activity-item/index.test.js]: Can we use `{...activity}` here?

Comment (adamweeks) [inline packages/widget-chat/src/components/activity-item/index.test.js]: Yes. Yes we can! 👍

**Comment by bzang** [comment]
👍  Looks good to me

**Pull request merged**